sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -97,6 +97,16 @@ object IntervalUtils {
     Decimal(result, 18, 6)
   }
 
+  def divide(interval: CalendarInterval, divisor: Decimal): CalendarInterval = {
+    if (divisor == Decimal.ZERO || divisor == null) return null
+    val months = Decimal(interval.months) / divisor
+    val days = months.remainder(Decimal.ONE) * Decimal(DAYS_PER_MONTH) +
+      Decimal(interval.days) / divisor
+    val milliseconds = days.remainder(Decimal.ONE) * Decimal(DateTimeUtils.MICROS_PER_DAY) +
+      Decimal(interval.microseconds) / divisor
+    new CalendarInterval(months.toInt, days.toInt, milliseconds.toLong)
```

Comment:
`milliseconds`?

Reply:
microseconds. thanks.

Reply:
this won't be needed when your pr will be merged. I'd like to leave it for a while.